### DISCUSSION TITLE
how not to close the connection?

### DISCUSSION THREAD

**Discussion by ilya-savichev** [bluenviron/mediamtx#723] [Questions]
When the ffmeg stream is not running, the client crashes with an error about a non-existent stream, this happens when the playback of files from the folder is broken. how can you keep the connection? there is some kind of timeout for waiting for ffmpeg to start?
(i want to output to "-f rtsp" the looped playback of files in a folder)
![image](https://user-images.githubusercontent.com/16954690/145216903-4f66018a-04f8-4a90-a463-f6c8dca8c984.png)
![image](https://user-images.githubusercontent.com/16954690/145217020-0a271545-e834-441b-9eb8-51761cb16280.png)

**Comment by aler9** [comment]
Hello, when a stream is re-initialized, the server kicks all existing clients, and all clients must be re-initialized, there's nothing else that can be done.

This is not a server choice but is caused by technical reasons. Each file can be encoded in a different way. In RTSP, the stream encoding is sent once during the initial handshake. There's no way to send it back again. Therefore, if it changes, the only way to apply this change is to close any existing connection.

You can avoid this by merging the two files together before sending them to the server; if they are both H264 and they use the same profile (baseline, main or high), you can convert them to MPEG-TS and concatenate them with ffmpeg, there are a lot of tutorials about how to do this.

[a comment by an automated account is not shown]
